scripts/gradle/gradle-check.sh

```diff
@@ -20,6 +20,39 @@ TRIGGER_TOKEN=$1
 PR_TITLE_NEW=`echo $pr_title | tr -dc '[:alnum:] ' | tr '[:upper:]' '[:lower:]'`
 PAYLOAD_JSON="{\"pr_from_sha\": \"$pr_from_sha\", \"pr_from_clone_url\": \"$pr_from_clone_url\", \"pr_to_clone_url\": \"$pr_to_clone_url\", \"pr_title\": \"$PR_TITLE_NEW\", \"pr_number\": \"$pr_number\"}"
 
+perform_curl_and_process_with_jq() {
+    local url=$1
+    local jq_filter=$2
+    local max_retries=$3
+    local count=0
+    local success=false
+
+    while [ $count -lt $max_retries ]; do
+        response=$(curl -s -XGET "${url}api/json")
+        curl_exit_code=$?
+
+        if [ $curl_exit_code -eq 0 ]; then
```

Comment:
We can just use jq to get status, as failed curl or 404 will just resulted in data that is not able to be parsed by jq.

Reply:
Thanks, Peter, agreed.. let me do that!